Only process xen image files with .gz extension and regular files
not symlinks.

--- util/grub.d/20_linux_xen.in.orig	2011-02-09 14:34:45.239824080 +0100
+++ util/grub.d/20_linux_xen.in	2011-02-09 14:36:26.760493023 +0100
@@ -98,8 +98,8 @@ linux_list=`for i in /boot/vmlinu[xz]-*
 	version=$(echo $basename | sed -e "s,^[^0-9]*-,,g")
         if grub_file_is_not_garbage "$i" && grep -qx "CONFIG_XEN_DOM0=y" /boot/config-${version} 2> /dev/null ; then echo -n "$i " ; fi
       done`
-xen_list=`for i in /boot/xen*; do
-        if grub_file_is_not_garbage "$i" ; then echo -n "$i " ; fi
+xen_list=`for i in /boot/xen*.gz; do
+        if grub_file_is_not_garbage "$i" && ! test -h "$i"; then echo -n "$i " ; fi
       done`
 prepare_boot_cache=
 
